modules/transactions/src/main/java/org/apache/ignite/internal/tx/TransactionInternalCheckedException.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx;
+
+import static org.apache.ignite.lang.ErrorGroup.errorGroupByCode;
+import static org.apache.ignite.lang.ErrorGroup.extractGroupCode;
+import static org.apache.ignite.lang.ErrorGroups.Transactions.TX_ERR_GROUP;
+
+import java.util.UUID;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * Base class for checked transaction exceptions.
+ */
+public class TransactionInternalCheckedException extends IgniteInternalCheckedException {
+    /**
+     * Creates a new exception with the error code.
+     *
+     * @param code Full error code.
+     */
+    public TransactionInternalCheckedException(int code) {
+        super(UUID.randomUUID(), code);
+        assert extractGroupCode(code) == TX_ERR_GROUP.code() : "Error code does not relate to transaction error group [code=" + code + ']';
```

Comment:
Offtop: it's confusing that we should have such assertions in exceptions, plenty of such assertions actually.